### DISCUSSION TITLE
kitty startup slow on my laptop

### DISCUSSION THREAD

**Discussion by AntonLydike** [kovidgoyal/kitty#3792] [Q&A] [answered]
I just switched to kitty on my main machine, and am now in the process of switching on my laptop. One thing I noticed immediately was how slow kitty started. I never noticed on my main machine and ran some benchmarks. Here are the results:

![image](https://user-images.githubusercontent.com/14147713/124360703-7637b980-dc2b-11eb-932c-3ff57ed67190.png)

For this test I removed my kitty config, so this is using the default config.

My laptop is running NixOS 21.05, and kitty 0.20.3 created by Kovid Goyal.

The weird thing is, that my main machine has no problem starting kitty: 
![image](https://user-images.githubusercontent.com/14147713/124360780-ed6d4d80-dc2b-11eb-9379-937e9c49bf38.png)

My main machine is also running nixos, but it's a little older. It uses kitty 0.18.3.

Does anyone have any idea?

**Comment by kovidgoyal** [comment, marked as answer]
Hard to say, use strace/gperf/some other profiling tool. Most likely it will be either GPU drivers or fontconfig causing the slowdown. Although if startup times are important to you I would suggest using --single-instance.

**Reply by AntonLydike** [reply]
It was indeed a broken GPU config iirc. Getting correct GPU support enabled fixed kitty crashing on startup.

**Comment by madduck** [comment]
Even with `--single-instance`, it takes what feels like a *long time* between me spawning kitty, and the Window with shell appearing. I am totally used to just hit the key combo to start a terminal, and then just to start typing without waiting, and that has worked with rxvt-unicode and Alacritty just fine. With kitty, which I otherwise love, I have to pause for half a second, and it is really hurting my motivation to stick with this. I just want the terminal to pop up and be ready, not load.

**Reply by AntonLydike** [reply]
For me it was broken GPU drivers, which cause Kitty to time out/break/fail when initialising a GPU context. Hope this helps!

**Reply by madduck** [reply]
It's not crashing for me, there is just a noticeable delay.

Or how would I find out if there might be a GPU problem? `--debug-gl` shows no problems whatsoever.

**Reply by AntonLydike** [reply]
kitty itself wasn't crashing for me - GPU initialization was and that made kitty either re-try a bunch on startup or run into some kind of time-out before continuing with a backup mode. I can't remember many details now. I think I used something like `strace` or `gperf` and looked at system logs in general to finally see a bunch of GPU-related issues appear which made me realize GPU support was broken. I would suggest you try out some other GPU-acceleration tests and see if they report any issues with the GPU...

**Reply by kovidgoyal** [reply]
--single-instance on a properly working system takes single digit milliseconds to spawn a new window. Make sure you are using the official kitty binaries not a distro package.

**Reply by madduck** [reply]
> --single-instance on a properly working system takes single digit milliseconds to spawn a new window. Make sure you are using the official kitty binaries not a distro package.

I tried the latest "binary bundle" and I see zero difference to the Debian distro package:

```
lotus:/tmp/madduck/bin% time ./kitty --single-instance true
./kitty --single-instance true  0,64s user 0,07s system 84% cpu 0,767 total
lotus:/tmp/madduck/bin% time kitty --single-instance true
kitty --single-instance true  0,59s user 0,09s system 86% cpu 0,721 total
lotus:/tmp/madduck/bin% time rxvt-unicode -e true
rxvt-unicode -e true  0,04s user 0,03s system 86% cpu 0,088 total
```

**Comment by kovidgoyal** [comment]
You need to leave the instance running for --single-instance to work.
Run

kitty --single-instance

then in another terminal run

time kitty --single-instance true

**Comment by madduck** [comment]
Ah, okay, this looks better.

I start Kitty here as `x-terminal-emulator` and that standard interface does not have `--single-instance`. Is it possible to configure in `kitty.conf` that I always want single-instance for all instances? I use a tiling window manager, so I don't need multiple terminals within a single Kitty window.
